aws-rds-customdbengineversion/src/main/java/software/amazon/rds/customdbengineversion/CreateHandler.java

```diff
@@ -0,0 +1,120 @@
+            RESOURCE_IDENTIFIER,
+            RESOURCE_ID_MAX_LENGTH
+    );
+
+    public CreateHandler() {
+        this(HandlerConfig.builder()
+                .backoff(BACKOFF_DELAY)
+                .build());
+    }
+
+    public CreateHandler(HandlerConfig config) {
+        super(config);
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<RdsClient> proxyClient,
+            final Logger logger) {
+
+        final ResourceModel model = request.getDesiredResourceState();
+
+        if (StringUtils.isNullOrEmpty(model.getEngineVersion())) {
+            model.setEngineVersion(DEFAULT_ENGINE_NAME_PREFIX +
+                    dbCustomEngineVersionIdentifierFactory.newIdentifier()
+                            .withStackId(request.getStackId())
+                            .withResourceId(request.getLogicalResourceIdentifier())
+                            .withRequestToken(request.getClientRequestToken())
+                            .toString());
+        }
+
+        final Tagging.TagSet allTags = Tagging.TagSet.builder()
+                .systemTags(Tagging.translateTagsToSdk(request.getSystemTags()))
+                .stackTags(Tagging.translateTagsToSdk(request.getDesiredResourceTags()))
+                .resourceTags(new HashSet<>(Translator.translateTagsToSdk(request.getDesiredResourceState().getTags())))
+                .build();
+
+        return ProgressEvent.progress(model, callbackContext)
+                .then(progress -> safeCreateCustomEngineVersion(proxy, proxyClient, progress, allTags))
+                .then(progress -> modifyAfterCreate(proxy, proxyClient, progress))
+                .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
+    }
+
+    private ProgressEvent<ResourceModel, CallbackContext> modifyAfterCreate(final AmazonWebServicesClientProxy proxy,
+                                                                            final ProxyClient<RdsClient> proxyClient,
+                                                                            final ProgressEvent<ResourceModel, CallbackContext> progress) {
+        if (shouldUpdateCustomEngineVersion(progress)) {
+            return Commons.execOnce(
+                    progress,
+                    () -> updateCustomEngineVersion(proxy, proxyClient, progress),
+                    CallbackContext::isModified,
+                    CallbackContext::setModified
+            );
+        }
+        return progress;
+    }
+
+    private boolean shouldUpdateCustomEngineVersion(final ProgressEvent<ResourceModel, CallbackContext> progress) {
+        return !CustomEngineVersionStatus.AVAILABLE.toString().equalsIgnoreCase(progress.getResourceModel().getStatus());
```

Comment:
Can you elaborate on the reasoning for this code? We only update CEV upon non-available status? Which status we expect to get here?